app/views/common_parameters/_form.html.erb

```diff
@@ -1,15 +1,11 @@
-<%= form_for @common_parameter do |f| %>
-    <%= base_errors_for @common_parameter %>
-    <%= text_f f, :name %>
-    <%= param_type_selector(f) %>
-    <%= textarea_f f, :value, :value => f.object.value_before_type_cast, :size => "col-md-8",
-                   :rows => line_count(f, :value),
-                   :input_group_btn => fullscreen_input,
-                   :class => "form-control no-stretch #{'masked-input' if @common_parameter.hidden_value?}" %>
-    <%= checkbox_f f, :hidden_value, :class => 'hidden_value_textarea_switch', :onchange => 'turn_textarea_switch()', :checked => f.object.hidden_value? %>
-
-   <input type="hidden" id="old" value="<%= @common_parameter.value %>" />
-   <input type="hidden" id="new" value="<%= @common_parameter.value %>" />
-
-   <%= submit_or_cancel f %>
-<% end %>
+<%
+  props = {
+    id: @common_parameter.id,
+    name: @common_parameter.name,
+    keyType: @common_parameter.key_type,
+    value: "#{@common_parameter.value_before_type_cast}",
+    isMasked: @common_parameter.hidden_value?,
+    isNew: @common_parameter.new_record?
+  }
+%>
+<%= react_component('CommonParameterForm', props) %>
```

Comment:
how about:

```suggestion
<%= react_component('CommonParameterForm', @common_parameter) %>
```

Reply:
But `@common_parameter` is `CommonParameter` object without parameters `isMasked` & `isNew`, so I can't do that.

Reply:
though in React component you can receive it through props as `new_record?` and then rename it to `isNew`.

since it's a full-page component, how about using it with react-router? the page will load much faster,

if you will choose this way, then we can do an API call to fetch `@common_parameter`
and with using selectors we could do something like:
`export const selectIsNew = () => selectAPIResponse(state, KEY).new_record?`

Reply:
> though in React component you can receive it through props as new_record? and then rename it to isNew.

To be honest I didn't know about it. I'll look into it & if it works I'll update the code as you suggest.

> since it's a full-page component, how about using it with react-router? the page will load much faster

I can do that, never had chance to work with router before, it's time to try it.

Reply:
So I tried as you suggested and TBH it just cause more issues:
```
<%= react_component('CommonParameterForm', @common_parameter) %>

In js:

need to disable lint for every not-camel-case variable (key_type vs keyType)
value is not @common_parameter.value_before_type_cast anymore (need to get it somewhere else)

and so on ...
```
Can I keep the `props` variable and pass it instead of `@common_parameter`? The additional logic in JS component just for handling all that stuff is not worth IMHO.